test/Mvc.Analyzers.Test/AddResponseTypeAttributeCodeFixProviderIntegrationTest.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Analyzer.Testing;
+using Microsoft.AspNetCore.Mvc.Analyzers.Infrastructure;
+using Microsoft.CodeAnalysis;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Analyzers
+{
+    public class AddResponseTypeAttributeCodeFixProviderIntegrationTest
+    {
+        private MvcDiagnosticAnalyzerRunner AnalyzerRunner { get; } = new MvcDiagnosticAnalyzerRunner(new ApiConventionAnalyzer());
+
+        private CodeFixRunner CodeFixRunner => CodeFixRunner.Default;
+
+        [Fact]
+        public Task CodeFixAddsStatusCodes() => RunTest();
+
+        [Fact]
+        public Task CodeFixAddsMissingStatusCodes() => RunTest();
+
+        [Fact]
+        public Task CodeFixWithConventionAddsMissingStatusCodes() => RunTest();
+
+        [Fact]
+        public Task CodeFixWithConventionMethodAddsMissingStatusCodes() => RunTest();
+
+        [Fact]
+        public Task CodeFixAddsSuccessStatusCode() => RunTest();
+
+        [Fact]
+        public Task CodeFixAddsFullyQualifiedProducesResponseType() => RunTest();
+
+        private async Task RunTest([CallerMemberName] string testMethod = "")
+        {
+            // Arrange
+            var project = GetProject(testMethod);
+            var controllerDocument = project.DocumentIds[0];
+
+            var expectedOutput = Read(testMethod + ".Output");
+
+            // Act
+            var diagnostics = await AnalyzerRunner.GetDiagnosticsAsync(project);
+            var actualOutput = await CodeFixRunner.ApplyCodeFixAsync(
+                new AddResponseTypeAttributeCodeFixProvider(),
+                project.GetDocument(controllerDocument),
+                diagnostics[0]);
+
+            Assert.Equal(expectedOutput, actualOutput, ignoreLineEndingDifferences: true);
+        }
+
+        private Project GetProject(string testMethod)
+        {
+            var testSource = Read(testMethod + ".Input");
+            return DiagnosticProject.Create(GetType().Assembly, new[] { testSource });
+        }
+
+        private string Read(string fileName)
```

Comment:
Will replacing baselines be a thing for this? If not, then it's good as is.

Reply:
So the plan is to ship this and revise this based on preview1 feedback. It's sort of intentionally lightly (unit) tested to see if this approach sticks \ if we need to take a different approach to providing these code fixes etc. I didn't want to too much infrastructure to support it at this point. But yeah, we would want to do the generate base lines thing at some point